dias/analyzers/sensitivity_analyzer.py

```diff
@@ -0,0 +1,273 @@
+            self.logger.error("No files match your search criteria. Exiting...")
+            return
+
+        #Get Unix time for the start time for timestamp
+        time_tuple      = start_time.timetuple()
+        start_time_unix = calendar.timegm(time_tuple)
+        timestamp0      = start_time_unix
+            
+        for files in all_files:
+        
+            t0 = time.time()
+            
+            # Look up inputmap
+            inputmap   = tools.get_correlator_inputs(ephemeris.unix_to_datetime(timestamp0),
+                                                   correlator=self.correlator)
+            # Load index map and reverse map
+            data       = andata.CorrData.from_acq_h5(files, 
+                                               datasets=['reverse_map', 'flags/inputs'],
+                                               apply_gain=False, renormalize=False)
+            
+            # Determine axes
+            nfreq     = data.nfreq
+            nblock    = int(np.ceil(nfreq / float(self.nfreq_per_block)))
+            
+            timestamp = data.time
+            ntime     = timestamp.size
+
+            # Get baselines
+            prod, prodmap, dist, conj, cyl, scale = self.get_baselines(data.index_map, inputmap)
+
+            # Determine groups
+            polstr = np.array(sorted(prod.keys()))
+            npol   = polstr.size
+            
+            t0 = time.time()
+            
+            # Calculate counts
+            cnt = np.zeros((data.index_map['stack'].size, ntime), dtype=np.float32)
+            
+            if np.any(data.flags['inputs'][:]):
+                for pp, ss in zip(data.index_map['prod'][:], data.reverse_map['stack']['stack'][:]):
+                    cnt[ss, :] += (data.flags['inputs'][pp[0], :] * data.flags['inputs'][pp[1], :])
+            else:
+                for ss, val in Counter(data.reverse_map['stack']['stack'][:]).iteritems():
+                    cnt[ss, :] = val
+            
+            self.logger.info("Took %0.1f seconds to get the counts" % (time.time() - t0))
+
+            # Initialize arrays
+            var 	= np.zeros((nfreq, npol, ntime), dtype=np.float32)
+            counter = np.zeros((nfreq, npol, ntime), dtype=np.float32)
+
+            # Loop over frequency blocks
+            for block_number in range(nblock):
+        
+                t0 = time.time()
+        
+                fstart   = block_number * self.nfreq_per_block
+                fstop    = min((block_number + 1) * self.nfreq_per_block, nfreq)
+                freq_sel = slice(fstart, fstop)
+        
+                self.logger.info("Processing block %d (of %d):  %d - %d" % (block_number+1, nblock, fstart, fstop))
+
+                bdata    = andata.CorrData.from_acq_h5(files, freq_sel=freq_sel,
+                                                    datasets=['flags/vis_weight'],
+                                                    apply_gain=False, renormalize=False)
```

Comment:
Have you tried loading the whole file in memory outside of the loop? It might make this much faster.
If it's done like now, make sure that this line doesn't load the whole file but only the selected frequencies into memory.

Reply:
It seems loading all frequencies leads to memory error. Unfortunately, we are interested in reading all the frequency channels for this analyzer so we do it in blocks.

Reply:
Okay. In that case, make sure the attribute `ondisk` is set to True (see [andata](https://bitbucket.org/chime/ch_util/src/master/ch_util/andata.py?at=master)). The difference between loading the whole file into memory before accessing the data and only loading the accessed data into memory can be huge.

I think if you used `andata.CorrReader`, that would set ondisk to True, but you can't turn off `renormalize` and `apply_gains`...

Reply:
I am not sure about this one, esp. with limited options. As of now, the processing time is faster, and it seems like it's possible to read it in blocks? I would soon push commits. Let me know once you try it on marimba.

Reply:
You should be able to test it on marimba yourself. Instructions are [here](https://dias.readthedocs.io/en/latest/dev/analyzer.html#testing-analyzers).

Reply:
Wow, thanks!

Reply:
Actually, I did follow these steps (I have an installed version of dias). The output seems fine, and each file is taking ~30 min to process. I checked these are not being overwritten now.